test/integration/targets/ecs_tag/tasks/full_test.yml

```diff
@@ -0,0 +1,350 @@
+        cluster_name: "{{resource_prefix}}"
+        resource: "{{resource_prefix}}"
+        resource_type: cluster
+        state: present
+        tags:
+          Name: "{{ resource_prefix }}"
+          another: foobar
+        <<: *aws_connection_info
+      register: taglist
+
+    - name: cluster tags - tags should be there
+      assert:
+        that:
+          - taglist.changed == true
+          - taglist.added_tags.Name == "{{ resource_prefix }}"
+          - taglist.added_tags.another == "foobar"
+
+    - name: cluster tags - Add tags to cluster again
+      ecs_tag:
+        cluster_name: "{{resource_prefix}}"
+        resource: "{{resource_prefix}}"
+        resource_type: cluster
+        state: present
+        tags:
+          Name: "{{ resource_prefix }}"
+          another: foobar
+        <<: *aws_connection_info
+      register: taglist
+
+    - name: cluster tags - No change after dding again
+      assert:
+        that:
+          - taglist.changed == false
+
+    - name: cluster tags - List tags
+      ecs_tag:
+        cluster_name: "{{ resource_prefix}}"
+        resource: "{{ resource_prefix}}"
+        resource_type: cluster
+        state: list
+        <<: *aws_connection_info
+      register: taglist
+
+    - name: cluster tags - should have 2 tags
+      assert:
+        that:
+          - taglist.tags|list|length == 2
+          - taglist.failed == false
+          - taglist.changed == false
+
+    - name: cluster tags - remove tag another
+      ecs_tag:
+        cluster_name: "{{resource_prefix}}"
+        resource: "{{resource_prefix}}"
+        resource_type: cluster
+        state: absent
+        tags:
+          another:
+        <<: *aws_connection_info
+      register: taglist
+
+    - name: cluster tags - tag another should be gone
+      assert:
+        that:
+          - taglist.changed == true
+          - '"another" not in taglist.tags'
+
+    - name: cluster tags - remove tag when not present
+      ecs_tag:
+        cluster_name: "{{resource_prefix}}"
+        resource: "{{resource_prefix}}"
+        resource_type: cluster
+        state: absent
+        tags:
+          temp:
+          temp_two:
+        <<: *aws_connection_info
+      register: taglist
+      ignore_errors: yes
+
+    - name: cluster tags - check that there was no fail, but changed is false
+      assert:
+        that:
+          - taglist.failed == false
+          - taglist.changed == false
+
+
+    - name: cluster tags - invalid cluster name
+      ecs_tag:
+        cluster_name: "{{resource_prefix}}-foo"
+        resource: "{{resource_prefix}}-foo"
+        resource_type: cluster
+        state: absent
+        tags:
+          temp:
+          temp_two:
+        <<: *aws_connection_info
+      register: taglist
+      ignore_errors: yes
+
+    - name: cluster tags - Make sure invalid clustername is handled
+      assert:
+        that:
+          - taglist.failed == true
+          - taglist.changed == false
```

Comment:
Can you please add a assertion for the contents of the failure message, to be sure we're failing in the expected way/place?

Reply:
yup